src/scenes/AccessionsRouter/edit/QuantityModal.tsx

```diff
@@ -114,14 +114,16 @@ export default function QuantityModal(props: QuantityModalProps): JSX.Element {
       } else {
         setSubsetWeightError('');
       }
-    } else {
-      setSubsetWeightError('');
     }
 
-    if (quantityChanged && isByWeight && !record.subsetCount) {
-      setSubsetCountError(strings.REQUIRED_FIELD);
-    } else {
-      setSubsetCountError('');
+    if (isByWeight || isSubsetOpen) {
+      if (!record.subsetCount) {
+        setSubsetCountError(strings.REQUIRED_FIELD);
```

Comment:
This is the same initial if-check as above, should it just be added to the same conditional?

Reply:
I was removing onditions and didn't realize about that. Thanks!